src/context/viewer-context.tsx

```diff
@@ -73,25 +92,36 @@ export type CustomDisplay = {
   };
 };
 
+export interface UserScrollingState {
+  state: boolean;
+  timeout: number | undefined;
+}
+
 export interface ViewerContextStore {
   activeCanvas: string;
   activeManifest: string;
+  autoScroll?: AutoScrollOptions;
```

Comment:
Should this be here? Looks like it is duplicating the configOption.

Reply:
I made some assumptions regarding the use of `configOptions`, the rest of the `ViewerContextStore`, and `ViewerAction`:

- `ViewerContextStore.configOptions` – the config that came in from the `ViewerProvider`. I treat it as read-only in case we need a “reset to initial state” trigger somewhere
- `ViewerContextStore.autoScroll` – the _current_ autoScroll behavior, which may have been changed programatically (I took this "original state/current state" pattern from the duplication of `configOptions.informationPanel.open` and `informationOpen`)
- `ViewerAction.autoScroll` – the new autoscroll setting coming in as part of a dispatch to update `ViewerContextStore.autoScroll.enabled`. Maybe it should be a full `AutoScrollOptions` object instead of just the boolean so it can all be updated programmatically instead of just the `enabled` part.

If it's OK for the current state to just overwrite the original state, I have no problem using `configOptions.informationPanel.autoScroll` throughout.

Other options:
- `autoScrolling` is the indicator of whether a `Cue` is currently in the middle of scrolling itself into view. It needed to be a top-level bool within the state because nesting it with other `autoScroll` stuff would trigger too many dependencies when it changes.
- `userScrolling` stores whether there's a manual user scroll happening, as well as the reference ID of the `setTimeout` that will reset that value. We could probably just use the presence of the timeout's ID to determine truthiness (e.g., `userScrolling === undefined` replaces `!userScrolling.enabled`), but this was more explicit.

Reply:
You're correct in that `configOptions` is read only.

I guess I was misunderstanding why `autoScroll` was being set at the top of the context as well. I could see a scenario where a user does toggle this on/off. But in this case, I think it would just be a boolean and not the whole autoScroll object. I hope I'm getting that correct now.

Some of my confusion here is self inflicted as a lot of the naming conventions here at the top of the config need to be thought out a bit better. This isn't on you but we could set a new standard here and now for this type of state variable.

I think we could go forward with more descriptive props like:

 - `isAutoScrollEnabled`
 - `isAutoScrolling`
 - `isUserScrolling`
 
 AND
 
 - `isInformationOpen`

Reply:
OK, this makes sense to me now.